### PR TITLE
fix nil pointer in StateSectorPreCommitInfo

### PR DESCRIPTION
**Pull request by leviok**

### CODE REVIEW

**Comment by magik6k** [comment]
A more correct fix would be to make this method return a pointer, and return nil/null when precommit was not found

**Comment by leviok** [comment]
You are right. This is a careless fix.

**Review by magik6k — COMMENTED** [review]
Can you rebase on latest master to fix build?

**Comment by leviok** [comment]
@magik6k done

**Review by magik6k — APPROVED** [review]

File: node/impl/full/state.go [hunk 1]
```diff
@@ -792,6 +792,8 @@ func (a *StateAPI) StateSectorPreCommitInfo(ctx context.Context, maddr address.A
 	pci, err := stmgr.PreCommitInfo(ctx, a.StateManager, maddr, n, ts)
 	if err != nil {
 		return miner.SectorPreCommitOnChainInfo{}, err
+	} else if pci == nil {
+		return nil, xerrors.Errorf("precommit info is not exists")
```
Comment (magik6k) [inline node/impl/full/state.go:796]: This needs to be 
```suggestion
		return miner.SectorPreCommitOnChainInfo{}, xerrors.Errorf("precommit info is not exists")
```

**Review by Kubuxu — APPROVED** [review]

**Pull request merged**